google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineExecutor.java

```diff
@@ -45,8 +45,15 @@ public void onStart(Process process) {
   @Override
   public void cancel() {
     if (process != null) {
-      process.destroy();
-      task.onCancel();
+      // Only destroy the process and signal cancellation if the process hasn't exited
```

Comment:
For more detail - we keep track of all the deployments for cancellation because the cancellation happens at the "cloud" level - each cloud can have multiple simultaneous running deployments so we need a handle on them all when the user hits the stop button.